compiler/src/iree/compiler/Codegen/Common/VectorLayoutAnalysis.cpp

```diff
@@ -476,6 +476,27 @@ static void propagateLayoutToTransposeOp(
   update(result, changed);
 }
 
+static void propagateLayoutToContractionOp(
+    vector::ContractionOp contraction,
+    ArrayRef<const DistributionLayout *> operandLattices,
+    ArrayRef<DistributionLayout *> resultLattices,
+    std::function<void(DistributionLayout *, ChangeResult)> update) {
+  // Contraction has only one vector result.
+  DistributionLayout *result = resultLattices[0];
+  // Get the init value of the contraction.
+  const DistributionLayout *init = operandLattices[2];
+
+  // If result lattice already has a layout, we cannot do anything. We do not
+  // impose layout conflicts on results.
+  if (result->hasLayout()) {
```

Comment:
Why don't we want to add a layout conflict resolution here?

Reply:
We do conflict resolutions only during enforcement. Propagation functions are forward functions which put layout **to results**. Enforcement functions are backward functions which put layout **to operands**.

It's easier to put layout conflict resolutions for something that is a "use" i.e. an operand, because layouts are essentially defined by "uses".